addons/account/models/account_move_line.py

```diff
@@ -462,12 +462,13 @@ def _compute_currency_id(self):
     def _compute_name(self):
         for line in self.filtered(lambda l: l.move_id.inalterable_hash is False):
             if line.display_type == 'payment_term':
-                term_lines = line.move_id.line_ids.filtered(lambda l: l.display_type == 'payment_term') | line
+                term_lines = line.move_id.line_ids.filtered(lambda l: l.display_type == 'payment_term') | self.filtered(lambda l: l.display_type == 'payment_term' and l.move_id == line.move_id)
                 name = line.move_id.payment_reference or ''
                 if len(term_lines) > 1:
-                    index = term_lines._ids.index(line.id) + 1
-                    name = f'{name} installment #{index}'.lstrip()
-                line.name = name
+                    for idx, term in enumerate(term_lines.sorted(key=lambda l: l.date_maturity), 1):
+                        term.name = _('%s installment #%s', name, idx).lstrip()
```

Comment:
A compute function is not supposed to change records that are not in `self`, why is it needed?

If it is just for the sorting, you can add it when building `terms_by_move` or when reading `term_lines` before doing the `index` (which could also be improved with a dict if needed for the time complexity, but that would probably make it less readable and useful)

Reply:
the problem is deeper than that, the problem is that when sync dinamic lines gets called on a move that has already 1 debit lines (an invoice with a single installement payment term) when we change the term to the problematic term (the one with 3 installements) instead of removing all debit lines and recreating them, it edits the debit line already present (which prevents the compute from triggering) and creates 2 which triggers the compute

Reply:
Would something like this on `account.move` do the trick?
```python
    @api.onchange('invoice_payment_term_id')
    def _inverse_invoice_payment_term_id(self):
        self.line_ids._conditional_add_to_compute('name', lambda l: (
            l.display_type == 'payment_term'
        ))
```

Reply:
doesn't work either because sync dynamic lines doesn't happen onchange of payment terms but rather on save

Reply:
Ooops, forgot to say that you need to add `inverse='_inverse_invoice_payment_term_id'` on the declaration of the `invoice_payment_term_id` field

Reply:
the problem persists, dynamic lines doesn't happen onchange of payment terms but rather on save